departures.go

```diff
@@ -153,6 +171,50 @@ func main() {
 	}
 }
 
+func searchStations(name string) ([]station, error) {
+	var stations []station
+	err := getJSON(&stations, "https://2.bvg.transport.rest/locations?query=%s&poi=false&addresses=false", name)
+
+	return stations, err
+}
+
+func promptForStation(name string) (*station, error) {
+	stations, err := searchStations(name)
+	if err != nil {
+		return nil, fmt.Errorf("could not query stations")
+	}
+
+	if len(stations) == 0 {
```

Comment:
I think the one result you could also skip the rest of the function.

Reply:
I'm not sure I understand. There's an early return in the next line. Could you clarify.

Reply:
You are returning for `len(stations) == 0` (no results) but you could also return early for `len(stations) == 1` since the user has not choice. That way they could provide the precise station name to circumvent the selection.

Reply:
Ahh, I see what you mean. So if there's an exact match, i.e. `len(stations) == 1`, no need for the selection we used the matched station directly.

Reply:
updated